### PR TITLE
Fixed deleting of page used in ELSE route of another page

### PR DESCRIPTION
**Pull request by pricea1**
### What is the context of this PR?
Routing page crashed when page or section referred to in ELSE route is deleted

### How to review 
1. Create 3 pages
2. Create a page route and specify ELSE destination another page
3. Delete the ELSE destination page
4. ELSE destination of page route should now be changed to next page

Repeat for using Section pages as rule and else destinations

### CODE REVIEW

**Comment by pricea1** [comment]
[Deleting page when referred to in routing as else destination](https://trello.com/c/dPqAz4Kg/1222-deleting-page-when-referred-to-in-routing-as-else-destination)

File: eq-author-api/src/businessLogic/onPageDeleted.js [hunk 1]
```diff
@@ -8,6 +9,12 @@ const onPageDeleted = (ctx, removedPageId) => {
       return;
     }
 
+    const elseRoute = page.routing.else;
+    if (elseRoute.pageId === removedPageId) {
+      elseRoute.pageId = null;
```
Comment (samiwel) [inline eq-author-api/src/businessLogic/onPageDeleted.js]: Probably worth just ensuring that the sectionId is nulled out here as well.

```js
page.routing.else = {
  sectionId: null,
  pageId: null,
  logical: NEXT_PAGE
}

Comment (pricea1) [reply eq-author-api/src/businessLogic/onPageDeleted.js]: sectionId now also nulled

**Comment by pricea1** [comment]
> Could we handle deleted sections in the else in the same way as they currently suffer from the same issue as these pages?

Deleted sections now handled as else and rule destination

**Pull request merged**